pkg/cableengine/ipsec/ipsec.go

```diff
@@ -95,7 +95,20 @@ func (i *Engine) StartEngine(ignition bool) error {
 		forwardToSubPostroutingRuleSpec := []string{"-j", "SUBMARINER-POSTROUTING"}
 		ipt.AppendUnique("nat", "POSTROUTING", forwardToSubPostroutingRuleSpec...)
 		forwardToSubForwardRuleSpec := []string{"-j", "SUBMARINER-FORWARD"}
-		ipt.AppendUnique("filter", "FORWARD", forwardToSubForwardRuleSpec...)
+		rules, err := ipt.List("filter", "FORWARD")
+		if err != nil {
+			klog.Fatalf("error listing the rules in FORWARD chain: %v", err)
+		}
+		insertAt := 0
+		if rules[len(rules)-1] == "-A FORWARD -j REJECT --reject-with icmp-host-prohibited" {
```

Comment:
Thanks for reviewing @tpantelis. After some more testing, I noticed that using "Insert" can lead to duplicate entries when Submariner-Engine POD gets restarted. I'll modify the patch to avoid these duplicate entries and will consider the suggestion.

Just for notes:
If a Chain exists but has no rules, ipt.List(...) returns success and the len(rules) will be 1. The zeroth entry will be the default policy associated with the chain.